apis/infrastructure/v1beta1/byohost_types.go

```diff
@@ -32,6 +32,26 @@ type ByoHostSpec struct {
 	BootstrapSecret *corev1.ObjectReference `json:"bootstrapSecret,omitempty"`
 }
 
+// HostStatus is a set of details about the host platform.
+type HostStatus struct {
```

Comment:
Where is the OS version stored? Will it be part of the OS name itself?

Reply:
> Where is the OS version stored? Will it be part of the OS name itself?

Yes. 

I used [this ](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/469#issuecomment-1091154744)example from the issue as the reference though some of the fields were removed or renamed according to our needs:

```
  nodeInfo:
    architecture: arm64
    bootID: 140d092c-9a92-4c56-b841-e1392d499a59
    containerRuntimeVersion: containerd://1.5.10
    kernelVersion: 5.4.0-105-generic
    kubeProxyVersion: v1.23.4
    kubeletVersion: v1.23.4
    machineID: 83fb87a59e494f40b7d93ed2ff8f23e1
    operatingSystem: linux
    osImage: Ubuntu 21.10
    systemUUID: 83fb87a59e494f40b7d93ed2ff8f23e1
```